### PR TITLE
homework 2: covering Exceptions with pytest

### PR DESCRIPTION
**Pull request by yspryn**

### CODE REVIEW

File: tests/yspryn/test_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+def test_index_error():
+    a = list()
+    b = [1, 23, 'no']
+    # try:
+    #     b[1]
+    # except IndexError:
+    #     pass
+    # else:
+    #     raise AssertionError
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:12]: remove commented code

File: tests/yspryn/test_exceptions.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        b[5]
+
+def test_zero_division():
+    with pytest.raises(ZeroDivisionError):
+        a = 78/0
+
+
+def test_attribute_error():
+    with pytest.raises(AttributeError):
+        a = "hello world"
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:27]: move this line from with block, try to leave only part that really causes error

File: tests/yspryn/test_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    test_dict = {1: 10, 2: 20, 3: 30}
+    with pytest.raises(KeyError):
+        test_dict[4]
+
+
+def test_name_error():
+    my_list = [1, 2, 3, 55]
+    a1 = 10
+    with pytest.raises(NameError):
+        var = my_lists[1]
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:41]: no need in assignment operation

File: tests/yspryn/test_exceptions.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    with pytest.raises(NameError):
+        var = a11 + 6
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:43]: no need in assignment operation

File: tests/yspryn/test_exceptions.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    def test_syntax_error():
+        with pytest.raises(SyntaxError):
+            import my_test_module
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:48]: wrong identation,
this test never runs

File: tests/yspryn/test_exceptions.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+def test_stop_iteration():
+    with pytest.raises(StopIteration):
+        iterator = iter([1, 2, 3])
+        while True:
+            next_value = next(iterator)
+
+
+def test_import_error():
+    with pytest.raises(ImportError):
+       import non_existent_module
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:64]: as per pep8  empty line should be at the end of file

**Review by kudinov-fedor — CHANGES_REQUESTED** [review]

**Review by fkudinov-mn — APPROVED** [review]

File: tests/yspryn/test_exceptions.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import pytest
+import sys
+sys.path.append('./tests/yspryn')
```
Comment (kudinov-fedor) [inline tests/yspryn/test_exceptions.py:3]: remove this line

**Review by kudinov-fedor — APPROVED** [review]

**Pull request merged**